components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/dao/ResourceDAO.java

```diff
@@ -54,182 +56,138 @@
-
-            if (!resultSet.next()) {
-                throw new UMAClientException(UMAConstants.ErrorMessages
-                        .ERROR_CODE_NOT_FOUND_RESOURCE_ID, "Resource id : " + resourceid + " not found.");
-            } else {
-                resourceRegistration = new Resource();
-                do {
-                    if (!resourceRegistration.getScopes().contains(resultSet.getString(5))) {
-                        resourceRegistration.getScopes().add(resultSet.getString(5));
-                    }
-
-                    if (resultSet.getString(1) != null) {
-                        resourceRegistration.setResourceId(resultSet.getString(1));
-                    }
-                    if (resultSet.getString(2) != null) {
-                        resourceRegistration.setName(resultSet.getString(2));
-                    }
-                    if (resultSet.getString("PROPERTY_KEY").equals(ICON_URI)) {
-                        resourceRegistration.setIconUri(resultSet.getString("PROPERTY_VALUE"));
-                    }
-                    if (resultSet.getString("PROPERTY_KEY").equals(TYPE)) {
-                        resourceRegistration.setType(resultSet.getString("PROPERTY_VALUE"));
-                    }
-                    if (resultSet.getString("PROPERTY_KEY").equals(DESCRIPTION)) {
-                        resourceRegistration.setDescription(resultSet.getString("PROPERTY_VALUE"));
-                    }
-                } while (resultSet.next());
+    public static Resource retrieveResource(String resourceId) throws UMAServerException, UMAClientException {
+
+        Resource resource = new Resource();
+        NamedJdbcTemplate namedJdbcTemplate = JdbcUtils.getNewNamedTemplate();
+        Integer id;
+        try {
+            id = checkResourceExistence(resourceId);
+
+            resource.setResourceId(resourceId);
+
+            String name = namedJdbcTemplate.fetchSingleRecord(SQLQueries.GET_RESOURCE_NAME, (resultSet, i) ->
+                    resultSet.getString(1), namedPreparedStatement -> {
+                namedPreparedStatement.setString(UMAConstants.SQLPlaceholders.RESOURCE_ID, resourceId);
+            });
+            if (name != null) {
+                resource.setName(name);
             }
+
+            List<String> scopes = new ArrayList<>();
+            namedJdbcTemplate.executeQuery(SQLQueries.GET_RESOURCE_SCOPES, (resultSet, rowNumber) ->
+                    scopes.add(resultSet.getString(1)), namedPreparedStatement ->
+                    namedPreparedStatement.setInt(UMAConstants.SQLPlaceholders.ID, id));
+
+            resource.setScopes(scopes);
+
+            namedJdbcTemplate.executeQuery(SQLQueries.GET_RESOURCE_META_DATA, (resultSet, rowNumber) -> {
+                        if (resultSet.getString("PROPERTY_KEY").equals(ICON_URI)) {
+                            resource.setIconUri(resultSet.getString(
+                                    "PROPERTY_VALUE"));
```

Comment:
Use constants for "PROPERTY_KEY" and "PROPERTY_VALUE". Also, reuse those where needed.

Reply:
fixed: https://github.com/wso2-extensions/identity-oauth-uma/pull/68